pallets/spaces/src/lib.rs

```diff
@@ -181,12 +204,34 @@ decl_module! {
       let mut new_history_record = SpaceHistoryRecord {
         edited: WhoAndWhen::<T>::new(owner.clone()),
         old_data: SpaceUpdate {
+            parent_id: None,
             handle: None,
             content: None,
             hidden: None
         }
       };
 
+      // TODO: add tests for this case
+      if let Some(parent_id_opt) = update.parent_id {
+        if parent_id_opt != space.parent_id {
+
+          if let Some(parent_id) = parent_id_opt {
+            let parent_space = Self::require_space(parent_id)?;
+
+            Self::ensure_account_has_space_permission(
+              owner.clone(),
+              &parent_space,
+              SpacePermission::CreateSubspaces,
```

Comment:
Here we should check for two permissions: either `UpdateOwnSubspaces` or `UpdateAnySubspace`